File: src/main/java/at/searles/parsing/utils/common/ToInt.java
package at.searles.parsing.utils.common;

import at.searles.parsing.ParserCallBack;
import at.searles.parsing.Mapping;
import at.searles.parsing.ParserStream;
import at.searles.parsing.PrinterCallBack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ToInt implements Mapping<CharSequence, Integer> {
    @Override
    public Integer parse(ParserCallBack env, ParserStream stream, @NotNull CharSequence left) {
        try {
            return Integer.parseInt(left.toString());
        } catch (NumberFormatException e) {
            return null;
        }
    }

    @Nullable
    @Override
    public CharSequence left(PrinterCallBack env, @NotNull Integer result) {
        return result.toString();
    }
}

File: src/main/java/at/searles/parsingtools/common/ToInt.kt
package at.searles.parsingtools.common

import at.searles.parsing.Mapping
import at.searles.parsing.ParserStream

object ToInt : Mapping<CharSequence, Int> {
    override fun reduce(left: CharSequence, stream: ParserStream): Int {
        try {
            return left.toString().toInt()
        } catch (e: NumberFormatException) {
            error("input was not an int number")
        }
    }

    override fun left(result: Int): CharSequence {
        return result.toString()
    }

    override fun toString(): String {
        return "{x -> x.toInt()}"
    }
}
